### PR TITLE
Prometheus health check

### PR DESCRIPTION
**Pull request by arvind-chandra**
When using Thanos (a different flavor of Prometheus), the health check is "OK" This is not accounted for in the code, as it only checks for the string "Prometheus" and needs to be updated for this case.

#38425 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by arvind-chandra** [comment]
@architkulkarni Hey, would you be able to take a look at the pull request? Thanks!

**Comment by architkulkarni** [comment]
@arvind-chandra thanks for the contribution, can you add more details in the PR description and in the code comment about the flavor of prometheus that causes the issue?

Assigning to folks who might have more context on prometheus, feel free to reassign.

File: dashboard/modules/metrics/metrics_head.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ async def prometheus_health(self, req):
 
                 text = await resp.text()
                 # Basic sanity check of prometheus health check schema
-                if "Prometheus" not in text:
+                if "Prometheus" not in text or "OK" not in text:
```
Comment (rkooo567) [inline dashboard/modules/metrics/metrics_head.py:160]: can you add a comment to the code that explains this change?

Comment (arvind-chandra) [reply dashboard/modules/metrics/metrics_head.py:161]: @rkooo567 Added! Let me know if you have any other questions.

**Comment by rkooo567** [comment]
There's a lint faiure

**Comment by arvind-chandra** [comment]
@rkooo567 I ran the linter to fix the linting issues, so that should be good. I also ran some scripts to fix a commit that did not have a sign-off however I think I may have messed up and picked up a bunch of commits from main as well. Any suggestions on how to fix this would be appreciated!

**Comment by peterghaddad** [comment]
@rkooo567 I believe the lint error was resolved? Mind taking another poke for Arvind?

**Comment by peterghaddad** [comment]
@rkooo567 @architkulkarni is this good to go? Thanks!

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**